pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentDispatcherMultipleConsumers.java

```diff
@@ -191,6 +191,10 @@ public synchronized CompletableFuture<Void> addConsumer(Consumer consumer) {
         return CompletableFuture.completedFuture(null);
     }
 
+    protected synchronized void rewindCursor() {
```

Comment:
just wondering if it's necessary to make this method synchronized

Reply:
@lhotari 

>  just wondering if it's necessary to make this method synchronized

You're right, all the locations that called this method have already acquired the lock. The purpose of declaring `synchronized` here is to make the code more readable, and there is no performance overhead.

The point of locking is to prevent `add consumer` and `remove consumer` and `clear recentJoinedConsumers` execute concurrence.